ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
Readability wise, I prefer it returning early instead of wrapping the lot in a big `if` block. Makes it easier to follow IMO.